src/plugins/maps_legacy/public/map/base_maps_visualization.js

```diff
@@ -164,7 +164,7 @@ export function BaseMapsVisualizationProvider() {
 
       try {
         if (this._wmsConfigured()) {
-          if (WMS_MINZOOM > this._opensearchDashboardsMap.getMaxZoomLevel()) {
+          if (WMS_MAXZOOM > this._opensearchDashboardsMap.getMaxZoomLevel()) {
```

Comment:
Good catch! Thanks for the fix.

Reply:
Why do we need to also reset the min zoom level in this case (L168)? It seems like maybe the min and max zoom levels should each be reset under different conditions.

Reply:
Could we get more insight the original logic here and the original intent and ensure a regression isn't happening by changing this? Seems like code that has existed since even before the legacy version 6.x and it's changing the value.

From my understanding the original code was checking if the max zoom level was less than 0 then it would reset the min and max to 0 and 22.

Now the logic is if 22 > than max zoom level then it will always set the max zoom level to 22. So it seems regardless of what the map is configure to it will always set the max zoom level to 22.

Reply:
Hi @kavilla, There was a same issue reported at Kibana 7.1.1 at 2019 https://github.com/elastic/kibana/issues/49709. From that issue context, we could see the WMS zoom level limitation wasn't correct in some versions and didn't get fixed.

Based on this previous PR on Kibana https://github.com/elastic/kibana/commit/4e2401c216064ddc5fd9e7f5161db3e463b6c00a#diff-8eef4c226bf8f716c6420fcec4400202088971b418cb9bc31bdc4cecb57bc365R117, I still think it could be checking  with max zoom level with current existing max zoom level which is from opensearch maps service max zoom level.

If we don't set up the WMS_MAXZOOM to 22, the maps max zoom  would be set and limited by default maps service max zoom. So in order to make sure all WMS users are able to load their WMS tiles at max potential zoom level, we'd better set up  WMS_MAXZOOM to 22.

Hi @joshuarrrr, the code logic already exist since kibana, in my understand, here for WMS cases, it can make sure users always have zoom level from 0 - 22.

Reply:
Hi @junqiu-lei, yeah the bug makes sense in that it checks the maps service and defines the max zoom level which doesn't make sense if WMS is defined and it should use WMS configurations.

But I think this logic is intentional to be the case somehow that value returns as a negative and then it has some default case that at least set the same level from a valid (non-negative) value. But with change, regardless what the user configures for max zoom we will always set max zoom to 22. This seems like it addresses the bug as much is the increases the maximum to 22 instead of 10. It also prevents the edge case where that max zoom is a negative value. If somehow the value returned is negative for max zoom then max zoom would be negative.

So is the bug actually occurring at another location? Because I see earlier in the file the map gets configured with these default values and it must be overwritten, but in the rest of this function it sets the options that were configured so perhaps that value is not being applied or the previous base layer needs to be nulled out for WMS to take those settings?

Reply:
Hi @kavilla, I updated the WMS zoom check condition to include both existing check and bug fix check, so that we can  fix the bug and avoid abandon the existing logic.